util.go

```diff
@@ -9,6 +9,13 @@ import (
 	"time"
 )
 
+// ValidatorValuer is an interface that allows you to expose a method on a type
+// (including generic types) that returns a value that is supposed to be validated.
+type ValidatorValuer interface {
```

Comment:
I think we can drop the `Validator`, Outside this functions it's `validator.ValidatorValuer` which stutters.

`validator.Valuer` should be fine